core/src/env/arithmetic.rs

```diff
@@ -108,6 +110,16 @@ impl<T> BaseArithmetic for T where
         + DivAssign<Self>
         + CheckedMul
         + Saturating
+        + TryFrom<u16>
+        + TryFrom<u32>
+        + TryFrom<u64>
+        + TryFrom<u128>
+        + TryFrom<usize>
+        + TryInto<u16>
+        + TryInto<u32>
+        + TryInto<u64>
+        + TryInto<u128>
+        + TryInto<usize>
```

Comment:
Can you explain why we need this for this PR?

Reply:
Isn't this diverging from what Substrate bounds its types?

Reply:
> Can you explain why we need this for this PR?

The `TryFrom<u64>` is needed to convert the `gas: u64` arg into EnvTypes::Balance.

> Isn't this diverging from what Substrate bounds its types?

It's converging, I thought I would add the other `Try*` bounds while I was there: https://github.com/paritytech/substrate/blob/4be954a8463e6f33e0ec854c4b8bcd940af260fc/primitives/arithmetic/src/traits.rs#L44